### PR TITLE
SNOW-1830648: Fix nest query complexity adjustment

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1830648

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Update project complexity calculation by no directly updating dictionary which causes wrong updates.

### CODE REVIEW

File: src/snowflake/snowpark/_internal/analyzer/select_statement.py [hunk 1]
```diff
@@ -1002,9 +1002,10 @@ def projection_complexities(self) -> List[Dict[PlanNodeCategory, int]]:
                         dependent_column_complexity = (
                             subquery_projection_name_complexity_map[dependent_column]
                         )
-                        projection_complexity[PlanNodeCategory.COLUMN] -= 1
                         projection_complexity = sum_node_complexities(
-                            projection_complexity, dependent_column_complexity
+                            projection_complexity,
```
Comment (sfc-gh-yzou) [inline src/snowflake/snowpark/_internal/analyzer/select_statement.py:1006]: also update line 1000 to use proj.cumulative_node_complexity.copy() to be extra safe

**Review by sfc-gh-jdu — APPROVED** [review]

File: tests/integ/test_large_query_breakdown.py [hunk 2]
```diff
@@ -604,6 +607,31 @@ def test_optimization_skipped_with_views_and_dynamic_tables(session, caplog):
...
+
+    temp_table_name = Utils.random_table_name()
+    final_df = create_df_with_deep_nested_with_column_dependencies(
+        session, temp_table_name, 8
+    )
+
+    with SqlCounter(query_count=1, describe_count=0):
+        queries = final_df.queries
+    assert len(queries["queries"]) == 3
+    assert queries["queries"][0].startswith("CREATE  SCOPED TEMPORARY  TABLE")
```
Comment (sfc-gh-helmeleegy) [inline tests/integ/test_large_query_breakdown.py:624]: Why double spaces here?

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM overall, just one question/nit.

File: tests/integ/test_large_query_breakdown.py (hunk 2 above)
Comment (sfc-gh-aalam) [reply tests/integ/test_large_query_breakdown.py:624]: that's how the snowflake queries are generated. it is not always perfect. For example here
https://github.com/snowflakedb/snowpark-python/blob/8dd23318a9d9ce55aa116d4b443d405bb165516d/src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py#L827-L831

notice there is an additional space on line 828. Such spaces issue creep-in many places but it doesn't affect sql generation.

**Pull request merged**